launch.go

```
@@ -284,42 +91,43 @@ func getAvailableSchemas() []string {
 		"test_total_breakdown.json"}
 }
 
-func serveTemplate(w http.ResponseWriter, r *http.Request) {
-	lp := filepath.Join("templates", "layout.html")
-	fp := filepath.Join("templates", filepath.Clean(r.URL.Path))
+func getLaunchHandler(w http.ResponseWriter, r *http.Request) {
+	page := Page{Schemas: getAvailableSchemas()}
+	serveTemplate("launch.html", page, w, r)
+}
 
-	// Return a 404 if the template doesn't exist or is directory
-	info, err := os.Stat(fp)
-	if err != nil && (os.IsNotExist(err) || info.IsDir()) {
-		http.NotFound(w, r)
-		return
-	}
+func postLaunchHandler(w http.ResponseWriter, r *http.Request) {
+	var token string
+	var err error
 
-	tmpl, err := template.ParseFiles(lp, fp)
-	if err != nil {
-		log.Println(err.Error())
-		http.Error(w, http.StatusText(500), 500)
+	if err = r.ParseForm(); err != nil {
+		fmt.Fprintf(w, "POST. ParseForm() err: %v\n", err)
 		return
 	}
 
-	page := Page{Schemas: getAvailableSchemas()}
-	if err := tmpl.ExecuteTemplate(w, "layout", page); err != nil {
-		log.Println(err.Error())
-		http.Error(w, http.StatusText(500), 500)
+	if token, err = authentication.ConvertPostToToken(r.PostForm); err == nil {
+		hostUrl := settings.GetSetting("SURVEY_RUNNER_URL")
+		http.Redirect(w, r, hostUrl+"/session?token="+token, 301)
 	}
```

error isn't handled and silently fails with empty 200 response

---

now returns 500